fluster/system_info.py

```diff
@@ -0,0 +1,401 @@
+                    return value
+        return None
+
+    @staticmethod
+    def _parse_wmic_table(output: str, skip_header: str) -> Optional[str]:
+        """Parse wmic table format, skipping header and empty lines"""
+        lines = [item.strip() for item in output.split("\n") if item.strip()]
+        for line in lines[1:] if len(lines) > 1 else []:
+            if line and line != skip_header:
+                return line
+        return None
+
+    @staticmethod
+    def _bytes_to_gb(bytes_value: int) -> str:
+        """Convert bytes to GB with one decimal place"""
+        gb = bytes_value / (1024 * 1024 * 1024)
+        return f"{gb:.1f} GB"
+
+    def _get_cpu_model(self) -> str:
+        """Get CPU model information"""
+        try:
+            if self.os_name == "Linux":
+                with open("/proc/cpuinfo", encoding="utf-8") as f:
+                    for line in f:
+                        if "model name" in line:
+                            return line.split(":")[1].strip()
+            elif self.os_name == "Darwin":
+                output = self._run_command(["sysctl", "-n", "machdep.cpu.brand_string"])
+                if output:
+                    return output
+            elif self.os_name == "Windows":
+                output = self._run_command(["wmic", "cpu", "get", "name", "/format:list"])
+                if output:
+                    cpu_name = self._parse_wmic_list(output, "Name")
+                    if cpu_name:
+                        return cpu_name
+
+                output = self._run_command(["wmic", "cpu", "get", "name"])
+                if output:
+                    cpu_name = self._parse_wmic_table(output, "Name")
+                    if cpu_name:
+                        return cpu_name
+        except (FileNotFoundError, OSError, ValueError, IndexError):
+            pass
+        return "Unknown CPU"
+
+    @staticmethod
+    def _get_primary_gpu_pci() -> Optional[str]:
+        """Get the PCI ID of the primary GPU (Linux only)"""
+        try:
+            drm_path = "/sys/class/drm"
+            if not os.path.exists(drm_path):
+                return None
+
+            for device in os.listdir(drm_path):
+                if not device.startswith("card") or "-" in device:
+                    continue
+
+                boot_vga_file = os.path.join(drm_path, device, "device", "boot_vga")
+                if not os.path.exists(boot_vga_file):
+                    continue
+
+                with open(boot_vga_file, encoding="utf-8") as f:
+                    if f.read().strip() == "1":
+                        pci_link = os.path.join(drm_path, device, "device")
+                        if os.path.islink(pci_link):
+                            pci_path = os.readlink(pci_link)
+                            return pci_path.split("/")[-1]
+        except (OSError, ValueError, IndexError):
+            pass
+        return None
+
+    def _get_gpu_info(self) -> List[str]:
```

Comment:
I suggest to split the logic for each OS into its own function, it will be easier to read and more maintainable in the future.